test/integration/consul-container/test/ratelimit/ratelimit_test.go

```diff
@@ -0,0 +1,232 @@
+		},
+		httpAction:         "method=GET url=/v1/catalog/nodes",
+		rateLimitOperation: "Catalog.ListNodes",
+		rateLimitType:      "global/read",
+	}
+	putConfig := action{
+		function: func(client *api.Client) error {
+			_, err := utils.ApplyDefaultProxySettings(client)
+			return err
+		},
+		httpAction:         "method=PUT url=/v1/config",
+		rateLimitOperation: "ConfigEntry.Apply",
+		rateLimitType:      "global/write",
+	}
+
+	testCases := []testCase{
+		{
+			description: "Mode: disabled - errors: no / logs: no / metrics: no",
+			cmd:         "-hcl=limits { request_limits { mode = \"disabled\" read_rate = 0 write_rate = 0 }}",
+			operations: []operation{
+				{
+					action:           getNodes,
+					expectedErrorMsg: "",
+					expectLog:        false,
+					expectMetric:     false,
+				},
+				{
+					action:           putConfig,
+					expectedErrorMsg: "",
+					expectLog:        false,
+					expectMetric:     false,
+				},
+			},
+		},
+		{
+			description: "Mode: permissive - errors: no / logs: no / metrics: yes",
+			cmd:         "-hcl=limits { request_limits { mode = \"permissive\" read_rate = 0 write_rate = 0 }}",
+			operations: []operation{
+				{
+					action:           getNodes,
+					expectedErrorMsg: "",
+					expectLog:        false,
+					expectMetric:     false,
+				},
+				{
+					action:           putConfig,
+					expectedErrorMsg: "",
+					expectLog:        false,
+					expectMetric:     false,
+				},
+			},
+		},
+		{
+			description: "Mode: enforcing - errors: yes / logs: yes / metrics: yes",
+			cmd:         "-hcl=limits { request_limits { mode = \"enforcing\" read_rate = 0 write_rate = 0 }}",
+			operations: []operation{
+				{
+					action:           getNodes,
+					expectedErrorMsg: retryableErrorMsg,
+					expectLog:        true,
+					expectMetric:     true,
+				},
+				{
+					action:           putConfig,
+					expectedErrorMsg: nonRetryableErrorMsg,
+					expectLog:        true,
+					expectMetric:     true,
+				},
+			},
+		}}
+
+	for _, tc := range testCases {
+		t.Run(tc.description, func(t *testing.T) {
+			urlsExpectingLogging := []string{}
+			logConsumer := &TestLogConsumer{}
+			cluster := createCluster(t, tc.cmd, logConsumer)
+			defer terminate(t, cluster)
+
+			client, err := cluster.GetClient(nil, true)
+			require.NoError(t, err)
+			for _, op := range tc.operations {
+				if op.expectLog {
+					urlsExpectingLogging = append(urlsExpectingLogging, op.action.httpAction)
+				}
+
+				err = op.action.function(client)
+				if len(op.expectedErrorMsg) > 0 {
+					require.Error(t, err)
+					require.Equal(t, op.expectedErrorMsg, err.Error())
+				} else {
+					require.NoError(t, err)
+				}
+			}
+
+			// validate logs
+			for _, httpRequest := range urlsExpectingLogging {
+				found := false
+				for _, msg := range logConsumer.Msgs {
+					if strings.Contains(msg, httpRequest) {
+						found = true
+					}
+				}
+				require.True(t, found, fmt.Sprintf("Log not found for: %s", httpRequest))
```

Comment:
This sometimes occurs with `/v1/config PUT` where there is no logged messaged for it. Is there anything about `GetClient()` that would return a connection to a leader vs a follower that might cause this error to sometimes appear vs not?